src/storage/src/hummock/compactor/shared_buffer_compact.rs

```diff
@@ -100,12 +100,31 @@ pub async fn compact(
 /// For compaction from shared buffer to level 0, this is the only function gets called.
 async fn compact_shared_buffer(
     context: Arc<CompactorContext>,
-    payload: UploadTaskPayload,
+    mut payload: UploadTaskPayload,
 ) -> HummockResult<Vec<LocalSstableInfo>> {
     // Local memory compaction looks at all key ranges.
+
+    let mut existing_table_ids: HashSet<u32> = payload
+        .iter()
+        .map(|imm| imm.table_id.table_id)
+        .dedup()
+        .collect();
+
+    assert!(!existing_table_ids.is_empty());
+
+    let multi_filter_key_extractor = context
+        .filter_key_extractor_manager
+        .acquire(existing_table_ids.clone())
+        .await?;
+    if let FilterKeyExtractorImpl::Multi(multi) = &multi_filter_key_extractor {
+        existing_table_ids = multi.get_exsting_table_ids();
+    }
+    let multi_filter_key_extractor = Arc::new(multi_filter_key_extractor);
+
     let mut size_and_start_user_keys = vec![];
     let mut compact_data_size = 0;
     let mut builder = DeleteRangeAggregatorBuilder::default();
+    payload.retain(|imm| existing_table_ids.contains(&imm.table_id.table_id));
```

Comment:
Is this a low frequency event? do we need logs

Reply:
done